test/plugin/base_test.rb

```diff
@@ -1160,11 +1334,20 @@ def setup_prometheus
     Prometheus::Client.registry.instance_variable_set('@metrics', {})
   end
 
+  def setup_metadata_agent_stubs(resource_type)
```

Comment:
As I understand the way webmock works, we can stub out all the possible URLs at once (in fact, we could even have a static unique_id->monitored_resource map, and use [dynamic response generation](https://github.com/bblimke/webmock#responses-dynamically-evaluated-from-lambda)). We can then set expectations for specific URLs being requested in individual tests.

Reply:
Refactored. PTAL.